h2o-py/h2o/frame.py

```diff
@@ -108,6 +108,9 @@ def __init__(self, python_obj=None, destination_frame=None, header=0, separator=
             self._upload_python_object(python_obj, destination_frame, header, separator,
                                        column_names, column_types, na_strings, skipped_columns)
 
+    def __del__(self):
```

Comment:
IMHO this is not a good idea. You might want to import a file in Flow, then go to Python, use a reference to the object and then continue in Flow. I don't think (all) Frames in H2O should copy the same lifecycle as the client objects.

In R:
- when you create a temporary frame and the frame is garbage collected, corresponding H2O object is delete on the backend
- non-temporary objects are not deleted in the H2O backend

I think you change affects all H2OFrames.

IMHO this PR should (ideally) only address the reported issue. If you want to make a global change - I think it would be better to make a new jira and describe what was changed and why.

Reply:
In my opinion, we should use the FrameSplitter if it is possible (if there is API to use it from Python). However, I guess the reason why the Python code is used directly instead of Java is the speed, or not?

Reply:
Resolved, waiting for tests.

Reply:
FrameSplitter is currently not used at all except for tests, it was created for the old Python client.

Split Frame is implemented the same way in R, Python & Flow (= 3x the same thing).

Having a backend implementation would make it IMHO make more efficient, I agree.